workspaces/devurls.md

```diff
@@ -71,6 +71,22 @@ If, however, you named the dev URL `reactproject`, then your URL is
 If you access a dev URL that hasn't been created, Coder automatically adds it to
 your dev URL list on the dashboard and sets the access level to **Private**.
 
+## Programmatically accessing dev URLs
+
+If you need programmatic access to authenticated dev URLs (i.e., dev URLs with
+access levels set to **private**, **organization**, or **authorized users**),
+you can run the following in the terminal:
+
+```console
+# Generate an API token with the Coder CLI
+$ coder tokens create devurl
+<TOKEN>
+
+# Send HTTP requests to the dev URL using the devurl_session cookie
+$ curl --cookie "devurl_session=<TOKEN>" <dev-url>
```

Comment:
We might want to use a header here over a cookie, as I think cookies have CSRF protections applied. @Emyrk what do you think?

Reply:
Devurls do not have CSRF protection, as then it would likely break any app being served over the devurl.

The CSRF is after the devurl handler: https://github.com/cdr/m/blob/8ad3c9675faeeb6ad82f1607e08d905b6a3abe0a/product/coder/service/coderd/routes.go#L52-L62

So the cookie is fine, although a header still might be cleaner

Reply:
@Emyrk Nice, good to know, thank you for the correction. Do we support devurl sessions as a header? If so, what's the header name?

Reply:
We don't support headers currently because the browser was the only workflow being considered. It'd be pretty easy to add header support though if we need to.

Reply:
@sreya Cool. Well if a cookie is fine from a security perspective, then I guess there's no reason to do a header :man_shrugging: